envoy/stream_info/filter_state.h

```diff
@@ -15,6 +16,13 @@
 namespace Envoy {
 namespace StreamInfo {
 
+class FilterStateInlineMapScope {
```

Comment:
I thought we evolved that it's better not to bother with Scope since it doesn't really give us any type safety unless we are willing to pollute the map with statics.

Reply:
I think this still be useful. It will be used as identify of cross-modules registries. See https://github.com/envoyproxy/envoy/pull/27928#discussion_r1243035242

Reply:
I think it's only used in the context of using your registry helper, which is not core to the usage model. However, you might want to use it in this case.